component/common/src/main/java/org/wso2/carbon/extension/identity/emailotp/common/EmailOtpServiceImpl.java

```diff
@@ -229,17 +294,19 @@ private ValidationResponseDTO isValid(SessionDTO sessionDTO, String emailOtp, St
         return new ValidationResponseDTO(userId, true);
     }
 
-    private SessionDTO issueOTP(User user) throws EmailOtpException {
+    private SessionDTO issueOTP(User user, int emailOtpExpiryTime) throws EmailOtpException {
 
         boolean triggerNotification = EmailOtpServiceDataHolder.getConfigs().isTriggerNotification();
         boolean resendSameOtp = EmailOtpServiceDataHolder.getConfigs().isResendSameOtp();
+        boolean isEnableMultipleSessions = EmailOtpServiceDataHolder.getConfigs().isEnableMultipleSessions();
 
         // If 'Resend same OTP' is enabled, check if such OTP exists.
         SessionDTO sessionDTO = null;
-        if (resendSameOtp) {
+        if (resendSameOtp && isEnableMultipleSessions== false) {
             sessionDTO = getPreviousValidOTPSession(user);
```

Comment:
shouldn't we retrieve the session with new session id (username + transactionId)?